Sources/SwiftProtobuf/Google_Protobuf_Any.swift

```diff
@@ -491,36 +494,52 @@ public struct Google_Protobuf_Any: Message, _MessageImplementationBase, _ProtoNa
-                    throw AnyUnpackError.malformedWellKnownTypeJSON
+                try contentJSON.withUnsafeBytes { (bytes:UnsafePointer<UInt8>) in
+                    var scanner = JSONScanner(utf8Pointer: bytes,
+                                              count: contentJSON.count)
+                    let key = try scanner.nextQuotedString()
+                    if key != "value" {
+                        // The only thing within a WKT should be "value".
+                        throw AnyUnpackError.malformedWellKnownTypeJSON
+                    }
+                    try scanner.skipRequiredColon()  // Can't fail
+                    let value = try scanner.skip()
+                    if !scanner.complete {
+                        // If that wasn't the end, then there was another key,
+                        // and WKTs should only have the one.
+                        throw AnyUnpackError.malformedWellKnownTypeJSON
+                    }
+                    // Note: This api is unpackTo(target:) so it really should be
+                    // a merge and not a replace (the non WKT case next is a merge).
+                    // The only WKTs where there would seem to be a difference are:
+                    //   Struct - It is a map, so it would merge into any existing
+                    //     enties.
+                    //   ValueList - Repeated, so values should append to the
+                    //       existing ones instead of instead of replace.
+                    //   FieldMask - Repeated, so values should append to the
+                    //       existing ones instead of instead of replace.
+                    //   Value - Interesting case, it is a oneof, so currently
+                    //       that would error if it was already set, so maybe
+                    //       replace is ok.
+                    target = try M(jsonString: value)
                 }
             } else {
-                // Decode JSON from the stored tokens for generated messages
-                guard let nameProviding = (target as? _ProtoNameProviding) else {
-                    throw JSONDecodingError.missingFieldNames
-                }
-                let fieldNames = type(of: nameProviding)._protobuf_nameMap
-                for (k,v) in jsonFields {
-                    if let fieldNumber = fieldNames.number(forJSONName: k) {
-                        let raw = v.data(using: String.Encoding.utf8)!
-                        try raw.withUnsafeBytes { (bytes: UnsafePointer<UInt8>) in
-                            var decoder = JSONDecoder(utf8Pointer: bytes, count: raw.count)
-                            try target.decodeField(decoder: &decoder, fieldNumber: fieldNumber)
-                            if !decoder.scanner.complete {
-                                throw JSONDecodingError.trailingGarbage
-                            }
-                        }
+                let asciiOpenCurlyBracket = UInt8(ascii: "{")
```

Comment:
Interesting.  I would have added the curly brackets when creating `_contentJSON`, but I guess that makes the WKT case a lot more complex?